src/react-components/ui-root.js

```diff
@@ -0,0 +1,608 @@
+    prefixMessageId: "entry.gearvr-prefix",
+    mediumMessageId: "entry.gearvr-medium"
+  };
+
+  return (<EntryButton  {...entryButtonProps}/>);
+};
+
+const DaydreamEntryButton = (props) => {
+  const entryButtonProps = {
+    ...props,
+    iconSrc: "./src/assets/images/daydream_entry.svg",
+    prefixMessageId: "entry.daydream-prefix",
+    mediumMessageId: "entry.daydream-medium"
+  };
+
+  return (<EntryButton  {...entryButtonProps}/>);
+};
+
+const AutoExitWarning = (props) => (
+  <div className="autoexit-panel">
+    <div className="autoexit-panel__title">
+      <FormattedMessage id="autoexit.title"/>
+      <span>{props.secondsRemaining}</span>
+      <FormattedMessage id="autoexit.title_units"/>
+    </div>
+    <div className="autoexit-panel__subtitle">
+      <FormattedMessage id="autoexit.subtitle"/>
+    </div>
+    <div className="autoexit-panel__cancel-button" onClick={props.onCancel}>
+      <FormattedMessage id="autoexit.cancel"/>
+    </div>
+  </div>
+);
+
+const ProfileInfoHeader = (props) => (
+  <div className="profile-info-header">
+    <img src="./src/assets/images/account.svg" onClick={props.onClick} className="profile-info-header__icon"/>
+    <div className="profile-info-header__profile_display_name" onClick={props.onClick}>
+      {props.name}
+    </div>
+    <div className="profile-info-header__app_name">
+      <b>moz://a</b> duck
+    </div>
+  </div>
+);
+
+// This is a list of regexes that match the microphone labels of HMDs.
+//
+// If entering VR mode, and if any of these regexes match an audio device,
+// the user will be prevented from entering VR until one of those devices is
+// selected as the microphone.
+//
+// Note that this doesn't have to be exhaustive: if no devices match any regex
+// then we rely upon the user to select the proper mic.
+const VR_DEVICE_MIC_LABEL_REGEXES = [];
+
+const AUTO_EXIT_TIMER_SECONDS = 10;
+
+class UIRoot extends Component {
+  static propTypes = {
+    enterScene: PropTypes.func,
+    availableVREntryTypes: PropTypes.object,
+    concurrentLoadDetector: PropTypes.object,
+    disableAutoExitOnConcurrentLoad: PropTypes.bool,
+    forcedVREntryType: PropTypes.string,
+    store: PropTypes.object,
+    scene: PropTypes.object
+  }
+
+  state = {
+    entryStep: ENTRY_STEPS.start,
+    enterInVR: false,
+
+    shareScreen: false,
+    mediaStream: null,
+
+    toneInterval: null,
+    tonePlaying: false,
+
+    micLevel: 0,
+    micDevices: [],
+    micUpdateInterval: null,
+
+    profileNamePending: "Hello",
+
+    autoExitTimerStartedAt: null,
+    autoExitTimerInterval: null,
+    secondsRemainingBeforeAutoExit: Infinity,
+
+    sceneLoaded: false,
+    exited: false,
+
+    showProfileEntry: false
+  }
+
+  componentDidMount() {
+    this.setupTestTone();
+    this.props.concurrentLoadDetector.addEventListener("concurrentload", this.onConcurrentLoad);
+    this.handleForcedVREntryType();
+    this.micLevelMovingAverage = MovingAverage(200);
```

Comment:
200 samples feels a bit too laggy IMO

Reply:
that's a timespan actually, 200ms. i'll try lowering it